### PR TITLE
- converted GHOrganization.Permission to a class

### PR DESCRIPTION
**Pull request by jgangemi**
potential fix for #1377.

i just converted the existing `enum` to a `class`. this only required minimal changes, so hopefully we can just roll w/ this change.

i'll fix the ci issues once it's decided if this is acceptable :)

### CODE REVIEW

**Comment by jgangemi** [comment]
@bitwiseman

File: src/main/java/org/kohsuke/github/GHOrganization.java [hunk 1]
```diff
@@ -448,10 +448,26 @@ public GHProject createProject(String name, String body) throws IOException {
     }
 
     /**
-     * The enum Permission.
+     * Permissions (roles) for teams and collaborators.
      */
-    public enum Permission {
-        ADMIN, MAINTAIN, PUSH, TRIAGE, PULL
+    public static class Permission {
+        // pre-canned permissions/roles
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHOrganization.java:454]: @jgangemi 

Sorry, no. We can't change the existing email to a class as that is a binary breaking change. We can create a new class but the existing one has to stay. 

We can add an `UNKOWN` value to the enum and add a `getPermissionString()` method to allow people to get the raw value. 

But we're not ready to make a breaking change at this time.

**Review by bitwiseman — CHANGES_REQUESTED** [review]
Can't make a breaking change like this.

File: src/main/java/org/kohsuke/github/GHOrganization.java (hunk 1 above)
Comment (jgangemi) [reply src/main/java/org/kohsuke/github/GHOrganization.java:454]: mm - right. forgot about that. all good, i'll do something else here.

**Pull request closed without merging**